src/PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/Toroidal.cpp

```diff
@@ -0,0 +1,114 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/Toroidal.hpp"
+
+#include <memory>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/InitialMagneticField.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace grmhd::AnalyticData::InitialMagneticFields {
+
+std::unique_ptr<InitialMagneticField> Toroidal::get_clone() const {
+  return std::make_unique<Toroidal>(*this);
+}
+
+Toroidal::Toroidal(CkMigrateMessage* msg) : InitialMagneticField(msg) {}
+
+void Toroidal::pup(PUP::er& p) {
+  InitialMagneticField::pup(p);
+  p | pressure_exponent_;
+  p | cutoff_pressure_;
+  p | vector_potential_amplitude_;
+}
+
+// NOLINTNEXTLINE
+PUP::able::PUP_ID Toroidal::my_PUP_ID = 0;
+
+Toroidal::Toroidal(const size_t pressure_exponent, const double cutoff_pressure,
+                   const double vector_potential_amplitude)
+    : pressure_exponent_(pressure_exponent),
+      cutoff_pressure_(cutoff_pressure),
+      vector_potential_amplitude_(vector_potential_amplitude) {}
+
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::MagneticField<DataType, 3>>
+Toroidal::variables(const tnsr::I<DataType, 3>& coords,
+                    const Scalar<DataType>& pressure,
+                    const Scalar<DataType>& sqrt_det_spatial_metric,
+                    const tnsr::i<DataType, 3>& dcoords_pressure) const {
+  auto magnetic_field = make_with_value<tnsr::I<DataType, 3>>(coords, 0.0);
+  const size_t num_pts = get_size(get(pressure));
+
+  for (size_t i = 0; i < num_pts; ++i) {
+    const double pressure_i = get_element(get(pressure), i);
+    if (pressure_i < cutoff_pressure_) {
+      get_element(magnetic_field.get(0), i) = 0.0;
+      get_element(magnetic_field.get(1), i) = 0.0;
+      get_element(magnetic_field.get(2), i) = 0.0;
+      continue;
+    }
+
+    // (p - p_c)^{n_s}
+    const double pressure_term =
+        pow(pressure_i - cutoff_pressure_, pressure_exponent_);
```

Comment:
you can multiply by `2.0` here already, right?

Reply:
yup!